pyproject.toml

```diff
@@ -91,6 +91,7 @@ youtube-transcript-api = "^0.6.1"
 beautifulsoup4 = "^4.12.2"
 pypdf = "^3.11.0"
 pytube = "^15.0.0"
+lxml = "^4.6.3" # required for sitemap xml parsing
```

Comment:
How come this dependency is needed all of the sudden? I don't see you import it.

Reply:
This is needed to allow beautiful soup to parse xml formatted files. Some sitemap.xml files are in xml format while others are in html format.

I was hoping this was going to be a quick fix but it seems that the sitemap implementation for this project is not complete. I'm going to pull this PR and submit it once I have a working implementation.